### PR TITLE
Use no-op global assembly cache helper when running under .net core

### PR DESCRIPTION
**Pull request by dibarbet**
Should fix https://github.com/dotnet/vscode-csharp/issues/6235

The `ClrGlobalAssemblyCache.cs` is used to find the implementation location of .net framework based assemblies that might be in the global assembly cache in order to decompile them.  

`ClrGlobalAssemblyCache.cs` is not valid to use as-is when running in a .net core runtime - it is not valid to [DllImport("clr")] - in fact this generally throws a dll not found exception (it gets caught and reported as a NFW).  I'm guessing in the above issue it managed to get a bit farther before throwing.

The solution here is to avoid calling `ClrGlobalAssemblyCache` when in .net core.  I replaced it with a no-op implementation.  While at some point it may be valuable to support decompiling .net framework assemblies in vscode, its not a high priority scenario and its not entirely clear on how that would be implemnted.

### CODE REVIEW

**Comment by dibarbet** [comment]
@tmat mind taking a look at this?  Since it looks like you were the last one to actually modify these

File: src/Compilers/Shared/GlobalAssemblyCacheHelpers/GlobalAssemblyCache.cs [hunk 1]
```diff
@@ -27,7 +25,9 @@ private static GlobalAssemblyCache CreateInstance()
             }
             else
             {
-                return new ClrGlobalAssemblyCache();
+                return System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.Contains(".NET Framework")
```
Comment (dibarbet) [inline src/Compilers/Shared/GlobalAssemblyCacheHelpers/GlobalAssemblyCache.cs:28]: if there is a preferred way to do this let me know.

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by dibarbet** [comment]
@dotnet/roslyn-compiler could use a couple reviews on the shared file changes

File: src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs
Comment (jjonescz) [inline src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs:1]: Why isn't this also under `src/Compilers/Shared/GlobalAssemblyCacheHelpers/`?

**Review by jjonescz — APPROVED** [review]

File: src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs
Comment (jaredpar) [reply src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs:1]: Agree we should keep the code in the same location.

File: src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/// <summary>
+/// Implements a no-op wrapper to search the global assembly cache when running under a .net core runtime.
+/// At some point we may wish to get information about assemblies in the GAC under a .net core runtime - for example
+/// if we're loading a .net framework project in vscode and want to find the actual assembly to decompile.
+/// 
+/// However it isn't extremely straightforward to implement and we don't need it at this time so leaving it as a no-op.
+/// More info on how this might be possible under a .net core runtime can be found https://github.com/dotnet/core/issues/3048#issuecomment-725781811
+/// </summary>
+internal class DotNetCoreGlobalAssemblyCache : GlobalAssemblyCache
```
Comment (jaredpar) [inline src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs:20]:
```suggestion
internal sealed class DotNetCoreGlobalAssemblyCache : GlobalAssemblyCache
```

File: src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs
Comment (dibarbet) [reply src/Scripting/Core/Hosting/Resolvers/DotNetCoreGlobalAssemblyCache.cs:1]: that's where VS automatically put it - I'll move it.

File: src/Compilers/Shared/GlobalAssemblyCacheHelpers/GlobalAssemblyCache.cs (hunk 1 above)
Comment (jaredpar) [reply src/Compilers/Shared/GlobalAssemblyCacheHelpers/GlobalAssemblyCache.cs:28]: Believe this is the preferred way. Pattern occurs several places in our code base.

File: src/Compilers/Shared/GlobalAssemblyCacheHelpers/DotNetCoreGlobalAssemblyCache.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/// <summary>
+/// Implements a no-op wrapper to search the global assembly cache when running under a .net core runtime.
+/// At some point we may wish to get information about assemblies in the GAC under a .net core runtime - for example
+/// if we're loading a .net framework project in vscode and want to find the actual assembly to decompile.
+/// 
+/// However it isn't extremely straightforward to implement and we don't need it at this time so leaving it as a no-op.
+/// More info on how this might be possible under a .net core runtime can be found https://github.com/dotnet/core/issues/3048#issuecomment-725781811
+/// </summary>
+internal sealed class DotNetCoreGlobalAssemblyCache : GlobalAssemblyCache
```
Comment (jaredpar) [inline src/Compilers/Shared/GlobalAssemblyCacheHelpers/DotNetCoreGlobalAssemblyCache.cs:20]: Should we just make this type a singleton vs. creating a new one every time?

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/Shared/GlobalAssemblyCacheHelpers/DotNetCoreGlobalAssemblyCache.cs (hunk 3 above)
Comment (dibarbet) [reply src/Compilers/Shared/GlobalAssemblyCacheHelpers/DotNetCoreGlobalAssemblyCache.cs:20]: Going to leave this as-is - this is already created as a singleton instance, as instantiated by the `GlobalAssemblyCache.CreateInstance` in the parent.

**Pull request merged**